functions/task4.js

```diff
@@ -0,0 +1,31 @@
+function divide(numerator, denominator) {
+  try {
+    if (typeof numerator !== "number" || typeof denominator !== "number") {
+      throw new Error("both arguments must be numbers");
+    }
+
+    if (denominator === 0) {
+      throw new Error("dividing by zero is not possible");
+    }
+
+    const result = numerator / denominator;
+
+    return result;
+  } catch (error) {
+    console.error(error);
```

Comment:
we don't need to see stack trace in this task so only error message will be enough
```js
console.error(error.message);
```